test/dnf/test_product_id.py

```diff
@@ -0,0 +1,97 @@
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+# along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+#
+# Red Hat trademarks are not licensed under GPLv2. No permission is
+# granted to use or replicate Red Hat trademarks that are incorporated
+# in this software or its documentation.
+#
+
+import unittest
+import os
+import tempfile
+from typing import Union
+
+from plugins.dnf.product_id import DnfProductManager
+
+
+class TestDnfPluginProductId(unittest.TestCase):
+
+    CACHE_FILE = "productid_repo_mapping.json"
+
+    def setUp(self) -> None:
+        self.temp_directory: Union[None, tempfile.TemporaryDirectory] = None
+
+    def tearDown(self) -> None:
+        if self.temp_directory is not None:
+            self.temp_directory.cleanup()
+
+    def _create_cache_file(self) -> str:
+        """
+        Create temporary directory and return path of cache file
+        """
+        self.temp_directory = tempfile.TemporaryDirectory()
+        return os.path.join(self.temp_directory.name, self.CACHE_FILE)
+
+    def _create_cache_file_read_only(self) -> str:
+        """
+        Create read-only temporary directory and return path of cache file
+        """
+        self.temp_directory = tempfile.TemporaryDirectory()
+        # Make directory read only
+        os.chmod(self.temp_directory.name, 0x400)
+        return os.path.join(self.temp_directory.name, self.CACHE_FILE)
+
+    def test_create_cache_file(self):
+        """
+        Test that it is possible to create cache file
+        """
+        cache_file = self._create_cache_file()
+        DnfProductManager._write_cache_file({"foo": "bar"}, cache_file)
+        data = DnfProductManager._read_cache_file(cache_file)
+        self.assertEqual(data, {"foo": "bar"})
+
+    def test_create_cache_file_in_read_only_directory(self):
+        """
+        Test that no exception is raised, when dnf plugin tries to create cache file
+        in directory that is read-only
+        """
+        cache_file = self._create_cache_file_read_only()
+        try:
+            DnfProductManager._write_cache_file({"foo": "bar"}, cache_file)
+        except PermissionError as err:
+            self.fail(f"DnfProductManager._write_cache_file(read_only_path) raised exception: {err}")
+
+    def test_read_cache_perm(self):
+        """
+        Test that no exception is raised, when dnf plugin tries to read cache file
+        that process cannot read due to restricted permissions
+        """
+        cache_file = self._create_cache_file()
+        DnfProductManager._write_cache_file({"foo": "bar"}, cache_file)
+        os.chmod(cache_file, 0x000)
+        try:
+            data = DnfProductManager._read_cache_file(cache_file)
+        except PermissionError as err:
+            self.fail(f"DnfProductManager._read_cache_file(read_only_path) raised exception: {err}")
```

Comment:
is the exception checking needed here? if an exception is raisen and the test does not expect any, the test will properly fail

Reply:
It is not necessary. I only wanted to emphasize that this exception should never happen.